pandas/core/groupby/groupby.py

```diff
@@ -1964,16 +1967,19 @@ def array_func(values: ArrayLike) -> ArrayLike:
                 # TODO: avoid special casing SparseArray here
                 if how in ["any", "all"] and isinstance(values, SparseArray):
                     pass
-                elif how in ["any", "all", "std", "sem"]:
+                elif alt is None or how in ["any", "all", "std", "sem"]:
                     raise  # TODO: re-raise as TypeError?  should not be reached
```

Comment:
is the comment "should not be reached" still accurate?

Reply:
Yes - I double checked by adding an `assert False` prior to raise and running the tests.